src/ExperimentUtils.py

```diff
@@ -0,0 +1,126 @@
+# File for making experiment functions.
+import time
+import torch
+import torch.nn.functional as F
+import numpy as np
+
+from torcheval.metrics.functional import multiclass_accuracy
+
+def train(model, train_loader, optimizer, device):
+    """
+    Function for training the model.
+
+    Arguments:
+        model: torch.nn.Module - model we want to train
+        train_loader: torch_geometric.loader.DataLoader - data wrapped up in a PyG loader
+        optimizer: torch.optim object - optimizer for gradient descent step
+        device: str - device on which to perform training
+    Return:
+        loss: float - loss on training  
+    """
+    model.train()
+    loss_all = 0
+
+    for data in train_loader:
+        data = data.to(device)
+        optimizer.zero_grad()
+        y_pred = model(data)
+        loss = F.cross_entropy(y_pred, data.y)
+        loss.backward()
+        loss_all += loss.item()
+        optimizer.step()
+    return loss_all / len(train_loader.dataset)
+
+def eval_accuracy(model, loader, device, num_classes):
+    """
+    Function for evaluating the model.
+
+    Arguments:
+        model: torch.nn.Module - model we want to evaluate
+        loader: torch_geometric.loader.DataLoader - data wrapped up in a PyG loader
+        device: str - device on which to perform evaluation
+        num_classes: int - number of classes in output space
+    Return:
+        error: float - error defined by specific metric  
+    """
+    model.eval()
+    error = 0
+
+    for data in loader:
+        data = data.to(device)
+        with torch.no_grad():
+            y_pred = model(data)
+            # Accuracy on a batch
+            error += multiclass_accuracy(torch.reshape(y_pred, (-1 ,num_classes)), torch.argmax(torch.reshape(data.y, (-1 ,num_classes)), dim = 1))
```

Comment:
Is this micro or macro averaged accuracy? You probably want to report both

Reply:
Also, there are a lot of reshapes and argmaxes, etc. happening in a single line --> Please refacotr into multiple lines to make the code more readable - this will also make the probability of bugs much lower.